### ISSUE TITLE
[RFC] Updating from Cartalyst/Sentry to Cartalyst/Sentinel

### ISSUE THREAD

**Issue by chartjes** [opencfp/opencfp#498]
Sentry has been deprecated, and using a deprecated library worries me. I did attempt to upgrade it but I got very unfocused and decided to also add test coverage while I did it.

The work did not require a lot of code changes but extensive DB schema changes were needed. These make me uncomfortable too.

The branch in question is ch-symfony-security-refactor for people who want to look at what I had been doing. I am unsure on how to proceed, looking for some advice and structure.

**Comment by mdwheele** [comment]
Mobile at the moment, so I can't comment in-depth. When I made an attempt at this, one of the most problematic issues was that resolving identity happens through multiple interfaces. Some controllers reference Sentry directly to grab the user and others use the IdentityProvider interface. 

Stubbing an IdentityProvider for tests seems much lighter than all the setup required to get Sentry/Sentinel to behave correctly. 

I think the following would be useful to continue:

1. Get all controllers using IdentityProvider to gate access to the User rather than grabbing Sentry/Sentinel directly from container. 
2. Create factories for test Users that can be used to simplify tests. 
3. Drop a lot of the setup from tests where we can. 

I check out the branch later and take a deeper look.

**Comment by chartjes** [comment]
@mdwheele 

I think an IdentityProvider would be a great first step -- creating a double for the provider rather than all the Sentry stuff seems like a much better testing approach

**Comment by mdwheele** [comment]
I'm checking this out; running through the upgrade now. I'll send some PRs to the branch shortly. Immediately ran into:

```
vagrant@homestead:~/Code/opencfp$ composer install


  [Seld\JsonLint\ParsingException]
  "./composer.lock" does not contain valid JSON
  Parse error on line 921:

  ---------------------^
  Expected: 'STRING' - It appears you have an extra trailing comma


install [--prefer-source] [--prefer-dist] [--dry-run] [--dev] [--no-dev] [--no-custom-installers] [--no-autoloader] [--no-scripts] [--no-progress] [--no-suggest] [-v|vv|vvv|--verbose] [-o|--optimize-autoloader] [-a|--classmap-authoritative] [--apcu-autoloader] [--ignore-platform-reqs] [--] [<packages>]...
```

**Comment by mdwheele** [comment]
I've opened #499 to fix a few issues I had getting the branch running for a pre-existing user. I've gone so far as to log in and that's it. I'll dig in a bit more now.

**Comment by mdwheele** [comment]
Working through a lot of test failures in the current branch. Going to try and merge `upstream/master` and start from that since the branch is somewhat behind.

**Comment by mdwheele** [comment]
Aight, I'm donezoes for tonight. The PR above has some basic fixes. I've got another branch on my fork where I rebased this branch onto `upstream/master`. It also fixes all failing tests. I need to verify that in my resolution of conflicts, I did not remove any improvements / break anything outside of test suite.

Comparison:

https://github.com/opencfp/opencfp/compare/ch-symfony-security-refactor...mdwheele:rebase/ch-security-refactor-onto-upstream-master

**Comment by mdwheele** [comment]
Based on conversation in #499, I've started from scratch. The work, compared against `upstream/master`, can be seen-in-progress at: https://github.com/opencfp/opencfp/compare/master...mdwheele:feature/use-identity-providers

I'm trying to do a good job of capturing incremental refactor steps per-commit for easier review. For instance, aac3fdb8a31bf64f4a0d9008d4903dbaf4529524 marks every place I found that is going to need to be refactored to use an implementation of `Authentication`, `IdentityProvider` or `AccountManagement`.

I'll send a PR as soon as the refactor is complete with tests passing. After that is merged, I'll implement `Sentinel` versions of those interfaces and swap them out in the container.
